website/src/components/props-table.tsx

```diff
@@ -35,6 +36,27 @@ const PropsTable = ({
     return null
   }
 
+  const themeComponent = theme.components[of]
+  const sizeValues = themeComponent?.sizes && Object.keys(themeComponent.sizes)
+  const variantValues =
+    themeComponent?.variants && Object.keys(themeComponent.variants)
+
+  /**
+   * If component has size prop, override the rendered values
+   * for `size` prop with  the component's size values
+   */
+  if (info.props.size && sizeValues) {
+    info.props.size.type.name = sizeValues
```

Comment:
When overriding the value with an array, we need to introduce the conditional rendering below (L89+) with `renderPropType`. What do you think about using 

```suggestion
    info.props.size.type.name = sizeValues.join(', ')
```

to concat all array values to a string?

Reply:
This makes sense and is cleaner. I'll update the PR to use this. Thanks @TimKolberger